test/lightning/projects/merge_projects_test.exs

```diff
@@ -2311,4 +2340,431 @@ defmodule Lightning.Projects.MergeProjectsTest do
+
+      source_job = build(:job, name: "job_a")
+
+      source =
+        build(:workflow)
+        |> with_trigger(source_trigger_webhook)
+        |> with_trigger(source_trigger_cron)
+        |> with_job(source_job)
+        |> with_edge({source_trigger_webhook, source_job})
+        |> insert()
+
+      target_trigger = build(:trigger, type: :webhook)
+      target_job = build(:job, name: "job_x")
+
+      target =
+        build(:workflow)
+        |> with_trigger(target_trigger)
+        |> with_job(target_job)
+        |> with_edge({target_trigger, target_job})
+        |> insert()
+
+      result = MergeProjects.merge_workflow(source, target)
+
+      # Should have 2 triggers (webhook maps to target, cron gets new ID)
+      assert length(result["triggers"]) == 2
+
+      # Webhook trigger should map to target
+      webhook_trigger = Enum.find(result["triggers"], &(&1["type"] == :webhook))
+      assert webhook_trigger["id"] == target_trigger.id
+
+      # Cron trigger should get a new ID (not mapped)
+      cron_trigger = Enum.find(result["triggers"], &(&1["type"] == :cron))
+      assert cron_trigger["id"] != source_trigger_cron.id
+      refute cron_trigger["delete"]
+    end
+
+    test "handles multiple candidates with identical bodies in non-final iteration" do
+      source_trigger = build(:trigger, type: :webhook)
+      source_job_a = build(:job, name: "job_a", body: "fn(state => state)")
+      source_job_b = build(:job, name: "job_b", body: "fn(state => state)")
+      source_job_c = build(:job, name: "job_c", body: "fn(state => state)")
+
+      source =
+        build(:workflow)
+        |> with_trigger(source_trigger)
+        |> with_job(source_job_a)
+        |> with_job(source_job_b)
+        |> with_job(source_job_c)
+        |> with_edge({source_trigger, source_job_a})
+        |> with_edge({source_trigger, source_job_b})
+        |> with_edge({source_trigger, source_job_c})
+        |> insert()
+
+      target_trigger = build(:trigger, type: :webhook)
+      # Three target jobs with same body (creates ambiguity)
+      target_job_x = build(:job, name: "job_x", body: "fn(state => state)")
+      target_job_y = build(:job, name: "job_y", body: "fn(state => state)")
+      target_job_z = build(:job, name: "job_z", body: "fn(state => state)")
+
+      target =
+        build(:workflow)
+        |> with_trigger(target_trigger)
+        |> with_job(target_job_x)
+        |> with_job(target_job_y)
+        |> with_job(target_job_z)
+        |> with_edge({target_trigger, target_job_x})
+        |> with_edge({target_trigger, target_job_y})
+        |> with_edge({target_trigger, target_job_z})
+        |> insert()
+
+      result = MergeProjects.merge_workflow(source, target)
+
+      # All jobs should be mapped (even with ambiguity)
+      # Algorithm will eventually map them all in final iteration
+      source_jobs = Enum.filter(result["jobs"], &(!&1["delete"]))
```

Comment:
@elias-ba I think this filter may be redundant and you can just assert on `result["jobs"]`?